wkcuber/wkcuber/compress.py

```diff
@@ -1,116 +1,61 @@
-        "target_path",
-        help="Output directory for the compressed WKW dataset.",
-        nargs="?",
-        default=None,
-        type=parse_path,
-    )
-
-    parser.add_argument(
-        "--layer_name",
-        "-l",
-        help="Name of the cubed layer (color or segmentation)",
-        default="color",
-    )
-
-    parser.add_argument(
-        "--mag", "-m", nargs="*", help="Magnification level", default=None
-    )
+"""This module takes care of compressing WEBKNOSSOS datasets."""
 
-    add_verbose_flag(parser)
-    add_distribution_flags(parser)
-
-    return parser
-
-
-def compress_mag(
-    source_path: Path,
-    layer_name: str,
-    target_path: Path,
-    mag: Mag,
-    args: Optional[Namespace] = None,
-) -> None:
-    Dataset.open(source_path).get_layer(layer_name).get_mag(mag).compress(
-        target_path=target_path, args=args
-    )
-
-
-def compress_mag_inplace(
-    target_path: Path, layer_name: str, mag: Mag, args: Optional[Namespace] = None
-) -> None:
-    Dataset.open(target_path).get_layer(layer_name).get_mag(mag).compress(args=args)
-
-
-def compress_mags(
-    source_path: Path,
-    layer_name: str,
-    target_path: Optional[Path] = None,
-    mags: Optional[List[Mag]] = None,
-    args: Optional[Namespace] = None,
+from argparse import Namespace
+from multiprocessing import cpu_count
+from pathlib import Path
+from typing import Annotated, Optional
+
+import typer
+from rich import print as rprint
+
+from webknossos import Dataset
+from webknossos.utils import get_executor_for_args
+from wkcuber.utils import DistributionStrategy
+
+app = typer.Typer(invoke_without_command=True)
+
+
+@app.callback()
+def main(
+    *,
+    target: Annotated[
+        Path,
+        typer.Argument(help="Path to your WEBKNOSSOS dataset.", show_default=False),
+    ],
+    jobs: Annotated[
+        int,
+        typer.Option(
+            help="Number of processes to be spawned.",
+            rich_help_panel="Executor options",
+        ),
+    ] = cpu_count(),
+    distribution_strategy: Annotated[
+        DistributionStrategy,
+        typer.Option(
+            help="Strategy to distribute the task across CPUs or nodes.",
+            rich_help_panel="Executor options",
+        ),
+    ] = DistributionStrategy.MULTIPROCESSING,
+    job_ressources: Annotated[
+        Optional[str],
+        typer.Option(
+            help='Necessary when using slurm as distribution strategy. Should be a JSON string \
+                (e.g., --job_resources=\'{"mem": "10M"}\')\'',
+            rich_help_panel="Executor options",
+        ),
+    ] = None,
 ) -> None:
-    if target_path is None:
-        target = source_path.with_suffix(".tmp")
-    else:
-        target = target_path
+    """Compress a given WEBKNOSSOS dataset."""
 
-    layer = Dataset.open(source_path).get_layer(layer_name)
-    if mags is None:
-        mags = list(layer.mags.keys())
+    rprint(f"Compressing [blue]{target}[/blue] ...")
```

Comment:
Other then color support, is there a reason to use `rprint` instead of `logging.info` etc?